sympy/functions/elementary/complexes.py

```diff
@@ -442,6 +442,18 @@ def _eval_rewrite_as_Abs(self, arg, **kwargs):
     def _eval_simplify(self, **kwargs):
         return self.func(factor_terms(self.args[0]))  # XXX include doit?
 
+    def _eval_as_leading_term(self, x, logx=None, cdir=0):
+        arg0 = self.args[0]
+        t = Dummy('t', positive=True)
+        if cdir == 0:
+            cdir = 1
+        z = arg0.subs(x, cdir*t)
+        if z.is_positive:
+            return S.One
+        elif z.is_negative:
+            return -S.One
+        else:
+            raise PoleError("Cannot expand %s around 0" % (self))
```

Comment:
Hmmm, well does that mean cases like `sign(I*x).as_leading_term(x)` should give a `PoleError` ?

I guess that's not required. Something like the above can return `I`

What I think should happen for functions like `sign` or `arg` etc is the following. So we can see that there is no real difference in what the series and the leading term should return for these functions.
So what we can do here is shift the code from `_eval_nseries` (which I did go through and think should work) to `_eval_as_leading_term` something like this 
```

    def _eval_as_leading_term(self, x, logx=None, cdir=0):
        arg0 = self.args[0]
        x0 = arg0.subs(x, 0)
        if x0 != 0:
            return self.func(x0)
        if cdir != 0:
            cdir = arg0.dir(x, cdir)
        return -S.One if re(cdir) < 0 else S.One     
```
and then we just use that in the _eval_nseries call, maybe something like this (you might need to verify !!!)
```
    def _eval_nseries(self, x, n, logx):
        c, _ = self.args[0].as_leading_term(x).as_coeff_exponent(x)
        return self.func(c.limit(x, 0))     
```
I think someone as simple as this would also do the job
```
    def _eval_nseries(self, x, n, logx, cdir=0):
        return self._eval_as_leading_term(x, logx=logx, cdir=cdir)
```